ultravox/data/types.py

```diff
@@ -40,15 +48,18 @@ class DatasetSplitConfig(helpers.Serializable):
     """Name of the split."""
     num_samples: int
     """Number of samples in the split"""
-    split_type: DatasetSplit = DatasetSplit.TRAIN
-    """Type of split, i.e., train or validation."""
+    split_type: Optional[DatasetSplit] = None
+    """Type of split, i.e., train, test, or validation."""
 
     def __post_init__(self):
-        """Automatically set is_validation if it's a validation split."""
-        if self.name == "test":
-            self.split_type = DatasetSplit.TEST
-        elif self.name == "validation":
-            self.split_type = DatasetSplit.VALIDATION
+        """Automatically set split type based on split name"""
+        if self.split_type is None:
+            if self.name == "test":
+                self.split_type = DatasetSplit.TEST
+            elif self.name == "validation":
```

Comment:
I updated the logic to require `self.split_type` to be explicitly defined if `self.name` is not "train", "validation", or "test". Otherwise, the value of `self.split_type` is inferred from `self.name`.

Reply:
What if we do "test" in self.name, and "validation" in self.name. In this scenario, if we have test.500, it will not recognize it is a test split. Or a more dangerous situation is, if we have something like validation.100, and we forget to include the datasetsplit, then it will default to train.

Reply:
Actually, i see you changed the code. I think it is fine actually. I think its good to merge!

Reply:
Right now, if we use names other than `train`, `validation`, and `test`, the `split_type` field has to be set specifically.